packages/flutter/lib/src/widgets/restoration.dart

```diff
@@ -0,0 +1,1003 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+
+import 'container.dart';
+import 'framework.dart';
+
+export 'package:flutter/services.dart' show RestorationBucket, RestorationId;
+
+/// Creates a new scope (namespace) for [RestorationId]s used by descendant
+/// widgets to claim [RestorationBucket]s.
+///
+/// {@template flutter.widgets.restoration.scope}
+/// A restoration scope inserts a [RestorationBucket] into the widget
+/// tree, which descendant widgets can access via [RestorationScope.of].
+/// It is uncommon for descendants to directly store data in this bucket. Instead, descendant
```

Comment:
tiny nit: might want to reflow some of these paragraphs (this line looks longer than 80 chars).